test/WebSites/ModelBindingWebSite/Controllers/FileUploadController.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace ModelBindingWebSite.Controllers
+{
+    public class FileUploadController : Controller
+    {
+        public FileDetails UploadSingle(IFormFile file)
+        {
+            if (!ModelState.IsValid)
+            {
+                return null;
+            }
+
+            var reader = new StreamReader(file.OpenReadStream());
+            var fileContent = reader.ReadToEnd();
+            ContentDispositionHeaderValue contentDisposition;
+            var parsedContentDisposition = ContentDispositionHeaderValue.TryParse(file.ContentDisposition,
+                                                                                  out contentDisposition);
+            var fileDetails = new FileDetails
+            {
+                Filename = contentDisposition.FileName,
+                Content = fileContent
+            };
+            return fileDetails;
+        }
+
+        public FileDetails[] UploadMultiple(IEnumerable<IFormFile> files)
+        {
+            if (!ModelState.IsValid)
+            {
+                return null;
+            }
```

Comment:
What's this protecting against? What code have you written for this feature that will result in invalid model state?